In [38]:
import pandas as pd
import numpy as np

In [39]:
data_path = '../data/raw_data/'
output_path = '../data/'
df = pd.read_csv(data_path+'직방크롤링원룸오피스텔.csv')
print(df.head())

    item_id   user_no sales_type sales_title service_type  \
0  33500027     77540         전세          전세         오피스텔   
1  33500123   1309339         월세          월세         오피스텔   
2  33500147  16519078         월세          월세         오피스텔   
3  33500206  17561097         전세          전세         오피스텔   
4  33500268  15882702         월세          월세         오피스텔   

                                              images  \
0  ['https://ic.zigbang.com/ic/items/33500027/1.j...   
1  ['https://ic.zigbang.com/ic/items/33500123/1.j...   
2  ['https://ic.zigbang.com/ic/items/33500147/1.j...   
3  ['https://ic.zigbang.com/ic/items/33500206/1.j...   
4  ['https://ic.zigbang.com/ic/items/33500268/1.j...   

                                  image_thumbnail  매매금액   보증금액   월세금액  ...  \
0  https://ic.zigbang.com/ic/items/33500027/1.jpg   NaN  13500    NaN  ...   
1  https://ic.zigbang.com/ic/items/33500123/1.jpg   NaN   1000   65.0  ...   
2  https://ic.zigbang.com/ic/items/33500147/1.jpg   NaN   2000

C:\Users\user\AppData\Local\Temp\ipykernel_24760\23400133.py:3: DtypeWarning: Columns (33,59,62) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv(data_path+'직방크롤링원룸오피스텔.csv')


In [40]:
VERSION = '1.0'

In [41]:
df['user_no'].nunique(), df['item_id'].nunique(), df['user_no'].isna().sum()

(1006, 82440, 0)

### 필요 데이터
> 부동산 매물 정보 : item_id, feature1, feeature2, ...

> infra 1 : x, y

> infra 2 : x, y

ex)

item_id, feature1, ... , featureN, infra_x1, infra_y1, infra_x2, infra_y2,...



### 기본 매물 정보 1

- 'item_id'

        매물 id : 82440개 (전체 데이터 수 82440개)
        등록순서대로 구성되었으므로 모두 고유한 값을 가짐
        
- 'user_no'

        유저 번호 : 1006개
        판매자(or 중개인) 아이디로 보면 될 것 같음

- 'sales_type'

        매물 유형
        ex) 전세, 월세

- 'sales_title'

        == sales_type

- 'service_type'

        건물 유형
        ex) 오피스텔

- 'images'

        해당 매물의 모든 이미지 url
        ex) 'https://ic.zigbang.com/ic/items/33500027/1.jpg', 'https://ic.zigbang.com/ic/items/33500027/2.jpg', ...

- 'image_thumbnail'

        해당 매물의 가장 첫 이미지 url
        ex) 'https://ic.zigbang.com/ic/items/33500027/1.jpg'

- '매매금액'

        매매금액
        단위 : 만원
        nan

- '보증금액'

        보증금액 : deposit
        단위 : 만원

- '월세금액'

        월세금액 : monthly_rent
        단위 : 만원

- '전용면적_m2'

        전용면적 : exclusive_area
        단위 : 제곱미터

- '공급면적_m2'

        공급면적 : supply_area
        단위 : 제곱미터

- '대지권면적_m2'

        대지권면적
        단위 : 제곱미터
        nan


In [42]:
df1 = df[['item_id', 'sales_type', 'service_type', '보증금액', '월세금액', '전용면적_m2', '공급면적_m2']].copy()
df1

item_id sales_type service_type   보증금액   월세금액  전용면적_m2  공급면적_m2
0      33500027         전세         오피스텔  13500    NaN    19.17    29.75
1      33500123         월세         오피스텔   1000   65.0    39.67    49.59
2      33500147         월세         오피스텔   2000  149.0    45.19   105.71
3      33500206         전세         오피스텔  25000    NaN    28.05    37.98
4      33500268         월세         오피스텔   2000  150.0    52.81   106.15
...         ...        ...          ...    ...    ...      ...      ...
82435  34726244         월세         오피스텔   2000  210.0    53.57    90.16
82436  34726267         월세         오피스텔   3700   76.0    19.80    28.76
82437  34726302         월세         오피스텔   1000   65.0    13.86    23.86
82438  34726304         월세         오피스텔    350  280.0    40.08    66.00
82439  34726359         월세         오피스텔   1000   65.0    29.12    53.22

[82440 rows x 7 columns]

### 기본 매물 정보 2

- 'address'

        구+동 정보

- 'jibunAddress'

        지번주소
        
- 'local1'

        시

- 'local2'

        구

- 'local3'

        동

- 'local4'

        지번이 들어가야될 거 같음 (추측)
        nan

- 'room_type_code'

        1,2,3,4,5
        뭔지 모르겠다

- 'room_type' 

        == room_type_code

- 'title'

        매물 추가 정보의 제목
        ex) 판매자가 직방에 설명해둔 상세 정보 탭의 제목

- 'status'

        상태정보
        open, closed, cancel
        무슨 의미인지는 다시 확인해볼 필요있음

- 'description'

        매물 추가 정보의 내용
        ex) 판매자가 직방에 설명해둔 상세 정보 탭의 내용

- 'secret_memo'

        nan

- 'is_zzim'

        nan

- 'random_location'

        위경도 정보


In [43]:
df2 = df[['item_id', 'jibunAddress', 'local1', 'local2', 'local3', 'room_type_code','status', 'random_location']].copy()
df2

item_id     jibunAddress local1 local2 local3  room_type_code  status  \
0      33500027       중구 황학동 787  서울특별시     중구    황학동               1  cancel   
1      33500123    구로구 구로동 650-4  서울특별시    구로구    구로동               3   close   
2      33500147        중구 흥인동 86  서울특별시     중구    흥인동               2   close   
3      33500206    도봉구 창동 630-10  서울특별시    도봉구     창동               4   close   
4      33500268        중구 흥인동 86  서울특별시     중구    흥인동               2   close   
...         ...              ...    ...    ...    ...             ...     ...   
82435  34726244   강남구 역삼동 831-45  서울특별시    강남구    역삼동               4    open   
82436  34726267    성북구 동선동1가 124  서울특별시    성북구  동선동1가               3    open   
82437  34726302  영등포구 당산동 121-43  서울특별시   영등포구    당산동               2    open   
82438  34726304      강남구 대치동 894  서울특별시    강남구    대치동               4    open   
82439  34726359   영등포구 여의도동 14-5  서울특별시   영등포구   여의도동               1    open   

                             random_location  
0       37.56736214187265,127.02296613394441  
1       37.49621059945249,126.87257656708832  
2        37.5662777574542,127.01694266740394  
3      37.643251120056284,127.03630435512244  
4       37.56627873736684,127.01697559113303  
...                                      ...  
82435   37.49389208067903,127.03072935050437  
82436  37.593366409494365,127.01872761305866  
82437   37.53088181386047,126.90751469005271  
82438   37.50411903197599,127.05444412993114  
82439   37.53038886115623,126.91979063028874  

[82440 rows x 8 columns]

### 기본 매물 정보 3

- 'parking'

        주차 가능 정보
        ex) '1대 가능', '불가능'

- 'elevator'

        엘레베이터 유무
        모두 True

- 'room_direction'

        매물 방향
        ex) 'W', 'SW', 'S', 'NW', 'SE', 'E', 'NE', 'N'

- 'movein_date'

        입주 가능 날짜 정보
        ex) '2022. 11. 19 이후 입주 가능', '2022. 10. 14 이후 입주 가능 (조정됨.)', ... , '즉시 입주 가능 (조정필요)'

- 'agent_comment'

        중개사 코멘트 정보

- 'pnu'

        ???

- 'floor'

        ???
        ex) '5', nan, '10', '12', '4', '2', '3', '9', '8', '옥탑방', ...

- 'floor_string'

        ???
        ex) '5', '고', '중', '10', '12', '저', '반지하', ...

- 'floor_all'

        매물이 있는 건물의 전체 층 수

- 'pets'

        애완동물 가능여부
        nan

- 'loan'

        대출 여부?? 저당권 여부??
        nan

- 'building_id'

        빌딩 아이디 : 2432개

- 'options'

        포함된 옵션
        ex) '01;02;03;05;08;10;11;12;', '01;02;03;04;10;11;12;', ...
        
- 'manage_cost'

        관리비
        단위 : 만원

- 'manage_cost_inc'

        관리비 포함 항목

- 'manage_cost_not_inc'

        관리비 불포함 항목

- 'is_realestate'

        all True

- 'is_room'

        방..?? 뭔소리지
        all False

- 'is_owner'

        집주인이 사냐 아니냐 그런건가
        all nan

- 'is_premium'

        프리미엄??
        nan, True

- 'is_homepage'

        홈페이지 유무??
        True, False



In [44]:
df3 = df[['item_id', 'parking', 'room_direction', 'movein_date', 'floor', 'floor_string', 'floor_all', 'building_id', 'options', 'manage_cost', 'manage_cost_inc','manage_cost_not_inc', 'is_homepage']].copy()
df3

item_id parking room_direction                         movein_date  \
0      33500027   1대 가능              W               2022. 11. 19 이후 입주 가능   
1      33500123   1대 가능             SW        2022. 10. 14 이후 입주 가능 (조정됨.)   
2      33500147   1대 가능              S  2022. 10. 04 이후 입주 가능 (~10/10까지평일)   
3      33500206   1대 가능             SW                            즉시 입주 가능   
4      33500268   1대 가능              S               2022. 09. 28 이후 입주 가능   
...         ...     ...            ...                                 ...   
82435  34726244   1대 가능              E                            즉시 입주 가능   
82436  34726267   1대 가능              N                     즉시 입주 가능 (조정필요)   
82437  34726302   1대 가능             NW                            즉시 입주 가능   
82438  34726304   1대 가능              S               2023. 01. 02 이후 입주 가능   
82439  34726359   1대 가능             SW               2023. 01. 08 이후 입주 가능   

      floor floor_string floor_all  building_id  \
0         5            5       18층         2698   
1       NaN            고       15층          688   
2       NaN            중       22층        18926   
3       NaN            중       14층        34006   
4        10           10       22층        18926   
...     ...          ...       ...          ...   
82435   NaN            저       15층        18653   
82436   NaN            고        7층        29919   
82437   7.0            7       15층        19484   
82438   NaN            저        7층        19372   
82439  11.0           11       13층         3234   

                              options  manage_cost manage_cost_inc  \
0            01;02;03;05;08;10;11;12;          9.0             04;   
1               01;02;03;04;10;11;12;         10.0          04;05;   
2         01;02;03;05;06;07;10;11;12;         13.0          04;05;   
3               01;02;03;05;10;11;12;          7.0          04;05;   
4            01;02;03;05;06;10;11;12;         18.0          04;05;   
...                               ...          ...             ...   
82435           01;02;03;05;10;11;12;         15.0             NaN   
82436        01;02;03;05;06;10;11;12;          9.0          04;05;   
82437  01;02;03;05;06;07;08;10;11;12;         13.0       03;04;05;   
82438  01;02;03;05;06;07;09;10;11;12;         18.0             NaN   
82439        01;02;03;04;09;10;11;12;         14.0          01;03;   

      manage_cost_not_inc  is_homepage  
0            01;02;03;05;         True  
1               01;02;03;         True  
2               01;02;03;         True  
3               01;02;03;         True  
4               01;02;03;         True  
...                   ...          ...  
82435     01;02;03;04;05;         True  
82436           01;02;03;         True  
82437              01;02;         True  
82438     01;02;03;04;05;         True  
82439           02;04;05;         True  

[82440 rows x 13 columns]

### 매물 기본 정보 4

- 'user_has_penalty'

        user == 판매자(집주인)?? 중개인??
        True, False

- 'building_type'

        nan

- 'room_gubun_code'

        2

- 'view_count'

        0

- 'updated_at'

        매물 정보 업데이트 일시

- 'is_first_movein'

        nan

- 'vr_key'

        nan

- 'vr_type_name'

        nan

- 'approve_date'

        승인날짜??
        ex) '2013년 9월13일', '20050414', '2002.10.22', ... -> 사용한다면 통일 필요

- 'bathroom_count'

        화장실 수
        1, 2, 3

- 'residence_type'

        거주행태 종류

- 'popular'

        array([nan, "{'title': '인기 있는 집입니다.', 'description': '지난 24시간 동안 주변에서 이 집에 관심을 보인\\n사용자가 급격하게 증가했습니다.', 'image_url': 'https://ic.zigbang.net/vp/upload/popular/pop_img_1.png'}"], dtype=object)
        매물의 인기도를 정성적으로 보여줄수 있는 척도가 될 수 있을 듯

- '자동종료대상'

        ?? -> 매물 정보 공개를 자동종료하는 매물인지 아닌지
        False, True

- '상태확인At'

        날짜

- '자동종료At'

        nan, 날짜

In [45]:
df['자동종료대상'].unique()

array([False,  True])

In [46]:
df4 = df[['item_id', 'user_has_penalty', 'bathroom_count', 'residence_type', '자동종료대상', '상태확인At', '자동종료At']].copy()
df4

item_id  user_has_penalty  bathroom_count residence_type  자동종료대상  \
0      33500027             False               1           오피스텔   False   
1      33500123             False               1           오피스텔   False   
2      33500147             False               1           오피스텔   False   
3      33500206             False               1           오피스텔   False   
4      33500268             False               1           오피스텔   False   
...         ...               ...             ...            ...     ...   
82435  34726244             False               1           오피스텔   False   
82436  34726267             False               1           오피스텔   False   
82437  34726302             False               1        도시형생활주택   False   
82438  34726304             False               1        도시형생활주택   False   
82439  34726359             False               1           오피스텔   False   

                         상태확인At 자동종료At  
0      2022-10-02T01:47:16.000Z    NaN  
1      2022-09-26T05:50:44.000Z    NaN  
2      2022-09-27T00:47:35.000Z    NaN  
3      2022-09-26T05:53:53.000Z    NaN  
4      2022-09-27T01:29:22.000Z    NaN  
...                         ...    ...  
82435  2022-12-28T11:58:34.000Z    NaN  
82436  2022-12-28T12:03:02.000Z    NaN  
82437  2022-12-28T12:15:35.000Z    NaN  
82438  2022-12-28T12:16:15.000Z    NaN  
82439  2022-12-28T12:26:42.000Z    NaN  

[82440 rows x 7 columns]

In [47]:
# df1 = df[['item_id', 'sales_type', 'service_type', '보증금액', '월세금액', '전용면적_m2', '공급면적_m2']].copy()
df2 = df[['item_id', 'jibunAddress', 'local1', 'local2', 'local3', 'room_type_code', 'status', 'random_location']].copy()
df3 = df[['item_id', 'parking', 'room_direction', 'movein_date', 'floor', 'floor_string', 'floor_all', 'building_id', 'options', 'manage_cost', 'manage_cost_inc','manage_cost_not_inc', 'is_homepage']].copy()
# df4 = df[['item_id', 'user_has_penalty', 'bathroom_count', 'residence_type', '자동종료대상', '상태확인At', '자동종료At']].copy()

df3

item_id parking room_direction                         movein_date  \
0      33500027   1대 가능              W               2022. 11. 19 이후 입주 가능   
1      33500123   1대 가능             SW        2022. 10. 14 이후 입주 가능 (조정됨.)   
2      33500147   1대 가능              S  2022. 10. 04 이후 입주 가능 (~10/10까지평일)   
3      33500206   1대 가능             SW                            즉시 입주 가능   
4      33500268   1대 가능              S               2022. 09. 28 이후 입주 가능   
...         ...     ...            ...                                 ...   
82435  34726244   1대 가능              E                            즉시 입주 가능   
82436  34726267   1대 가능              N                     즉시 입주 가능 (조정필요)   
82437  34726302   1대 가능             NW                            즉시 입주 가능   
82438  34726304   1대 가능              S               2023. 01. 02 이후 입주 가능   
82439  34726359   1대 가능             SW               2023. 01. 08 이후 입주 가능   

      floor floor_string floor_all  building_id  \
0         5            5       18층         2698   
1       NaN            고       15층          688   
2       NaN            중       22층        18926   
3       NaN            중       14층        34006   
4        10           10       22층        18926   
...     ...          ...       ...          ...   
82435   NaN            저       15층        18653   
82436   NaN            고        7층        29919   
82437   7.0            7       15층        19484   
82438   NaN            저        7층        19372   
82439  11.0           11       13층         3234   

                              options  manage_cost manage_cost_inc  \
0            01;02;03;05;08;10;11;12;          9.0             04;   
1               01;02;03;04;10;11;12;         10.0          04;05;   
2         01;02;03;05;06;07;10;11;12;         13.0          04;05;   
3               01;02;03;05;10;11;12;          7.0          04;05;   
4            01;02;03;05;06;10;11;12;         18.0          04;05;   
...                               ...          ...             ...   
82435           01;02;03;05;10;11;12;         15.0             NaN   
82436        01;02;03;05;06;10;11;12;          9.0          04;05;   
82437  01;02;03;05;06;07;08;10;11;12;         13.0       03;04;05;   
82438  01;02;03;05;06;07;09;10;11;12;         18.0             NaN   
82439        01;02;03;04;09;10;11;12;         14.0          01;03;   

      manage_cost_not_inc  is_homepage  
0            01;02;03;05;         True  
1               01;02;03;         True  
2               01;02;03;         True  
3               01;02;03;         True  
4               01;02;03;         True  
...                   ...          ...  
82435     01;02;03;04;05;         True  
82436           01;02;03;         True  
82437              01;02;         True  
82438     01;02;03;04;05;         True  
82439           02;04;05;         True  

[82440 rows x 13 columns]

In [48]:
df_basic = df[['item_id', 'service_type', 'residence_type', 'sales_type', '보증금액', '월세금액', '전용면적_m2', '공급면적_m2']].copy()
df_loc = df[['item_id', 'local1', 'local2', 'local3', 'jibunAddress', 'room_type_code','status', 'random_location', 'room_direction', 'floor', 'floor_string', 'floor_all', 'building_id']].copy()
df_option = df[['item_id', 'options', 'manage_cost', 'manage_cost_inc','manage_cost_not_inc']].copy()
# df_basic
# df_loc
df_option

item_id                         options  manage_cost manage_cost_inc  \
0      33500027        01;02;03;05;08;10;11;12;          9.0             04;   
1      33500123           01;02;03;04;10;11;12;         10.0          04;05;   
2      33500147     01;02;03;05;06;07;10;11;12;         13.0          04;05;   
3      33500206           01;02;03;05;10;11;12;          7.0          04;05;   
4      33500268        01;02;03;05;06;10;11;12;         18.0          04;05;   
...         ...                             ...          ...             ...   
82435  34726244           01;02;03;05;10;11;12;         15.0             NaN   
82436  34726267        01;02;03;05;06;10;11;12;          9.0          04;05;   
82437  34726302  01;02;03;05;06;07;08;10;11;12;         13.0       03;04;05;   
82438  34726304  01;02;03;05;06;07;09;10;11;12;         18.0             NaN   
82439  34726359        01;02;03;04;09;10;11;12;         14.0          01;03;   

      manage_cost_not_inc  
0            01;02;03;05;  
1               01;02;03;  
2               01;02;03;  
3               01;02;03;  
4               01;02;03;  
...                   ...  
82435     01;02;03;04;05;  
82436           01;02;03;  
82437              01;02;  
82438     01;02;03;04;05;  
82439           02;04;05;  

[82440 rows x 5 columns]

In [49]:
df_loc['Address'] = df_loc['local1'] +' '+ df_loc['jibunAddress']
df_loc

item_id local1 local2 local3     jibunAddress  room_type_code  status  \
0      33500027  서울특별시     중구    황학동       중구 황학동 787               1  cancel   
1      33500123  서울특별시    구로구    구로동    구로구 구로동 650-4               3   close   
2      33500147  서울특별시     중구    흥인동        중구 흥인동 86               2   close   
3      33500206  서울특별시    도봉구     창동    도봉구 창동 630-10               4   close   
4      33500268  서울특별시     중구    흥인동        중구 흥인동 86               2   close   
...         ...    ...    ...    ...              ...             ...     ...   
82435  34726244  서울특별시    강남구    역삼동   강남구 역삼동 831-45               4    open   
82436  34726267  서울특별시    성북구  동선동1가    성북구 동선동1가 124               3    open   
82437  34726302  서울특별시   영등포구    당산동  영등포구 당산동 121-43               2    open   
82438  34726304  서울특별시    강남구    대치동      강남구 대치동 894               4    open   
82439  34726359  서울특별시   영등포구   여의도동   영등포구 여의도동 14-5               1    open   

                             random_location room_direction floor  \
0       37.56736214187265,127.02296613394441              W     5   
1       37.49621059945249,126.87257656708832             SW   NaN   
2        37.5662777574542,127.01694266740394              S   NaN   
3      37.643251120056284,127.03630435512244             SW   NaN   
4       37.56627873736684,127.01697559113303              S    10   
...                                      ...            ...   ...   
82435   37.49389208067903,127.03072935050437              E   NaN   
82436  37.593366409494365,127.01872761305866              N   NaN   
82437   37.53088181386047,126.90751469005271             NW   7.0   
82438   37.50411903197599,127.05444412993114              S   NaN   
82439   37.53038886115623,126.91979063028874             SW  11.0   

      floor_string floor_all  building_id                Address  
0                5       18층         2698       서울특별시 중구 황학동 787  
1                고       15층          688    서울특별시 구로구 구로동 650-4  
2                중       22층        18926        서울특별시 중구 흥인동 86  
3                중       14층        34006    서울특별시 도봉구 창동 630-10  
4               10       22층        18926        서울특별시 중구 흥인동 86  
...            ...       ...          ...                    ...  
82435            저       15층        18653   서울특별시 강남구 역삼동 831-45  
82436            고        7층        29919    서울특별시 성북구 동선동1가 124  
82437            7       15층        19484  서울특별시 영등포구 당산동 121-43  
82438            저        7층        19372      서울특별시 강남구 대치동 894  
82439           11       13층         3234   서울특별시 영등포구 여의도동 14-5  

[82440 rows x 14 columns]

### 기본 DataFrame

In [50]:
df_basic

item_id service_type residence_type sales_type   보증금액   월세금액  전용면적_m2  \
0      33500027         오피스텔           오피스텔         전세  13500    NaN    19.17   
1      33500123         오피스텔           오피스텔         월세   1000   65.0    39.67   
2      33500147         오피스텔           오피스텔         월세   2000  149.0    45.19   
3      33500206         오피스텔           오피스텔         전세  25000    NaN    28.05   
4      33500268         오피스텔           오피스텔         월세   2000  150.0    52.81   
...         ...          ...            ...        ...    ...    ...      ...   
82435  34726244         오피스텔           오피스텔         월세   2000  210.0    53.57   
82436  34726267         오피스텔           오피스텔         월세   3700   76.0    19.80   
82437  34726302         오피스텔        도시형생활주택         월세   1000   65.0    13.86   
82438  34726304         오피스텔        도시형생활주택         월세    350  280.0    40.08   
82439  34726359         오피스텔           오피스텔         월세   1000   65.0    29.12   

       공급면적_m2  
0        29.75  
1        49.59  
2       105.71  
3        37.98  
4       106.15  
...        ...  
82435    90.16  
82436    28.76  
82437    23.86  
82438    66.00  
82439    53.22  

[82440 rows x 8 columns]

In [51]:
df_basic_2 = df_basic.merge(df_loc[['item_id','random_location']], on='item_id', how='left')
df_basic_2

item_id service_type residence_type sales_type   보증금액   월세금액  전용면적_m2  \
0      33500027         오피스텔           오피스텔         전세  13500    NaN    19.17   
1      33500123         오피스텔           오피스텔         월세   1000   65.0    39.67   
2      33500147         오피스텔           오피스텔         월세   2000  149.0    45.19   
3      33500206         오피스텔           오피스텔         전세  25000    NaN    28.05   
4      33500268         오피스텔           오피스텔         월세   2000  150.0    52.81   
...         ...          ...            ...        ...    ...    ...      ...   
82435  34726244         오피스텔           오피스텔         월세   2000  210.0    53.57   
82436  34726267         오피스텔           오피스텔         월세   3700   76.0    19.80   
82437  34726302         오피스텔        도시형생활주택         월세   1000   65.0    13.86   
82438  34726304         오피스텔        도시형생활주택         월세    350  280.0    40.08   
82439  34726359         오피스텔           오피스텔         월세   1000   65.0    29.12   

       공급면적_m2                        random_location  
0        29.75   37.56736214187265,127.02296613394441  
1        49.59   37.49621059945249,126.87257656708832  
2       105.71    37.5662777574542,127.01694266740394  
3        37.98  37.643251120056284,127.03630435512244  
4       106.15   37.56627873736684,127.01697559113303  
...        ...                                    ...  
82435    90.16   37.49389208067903,127.03072935050437  
82436    28.76  37.593366409494365,127.01872761305866  
82437    23.86   37.53088181386047,126.90751469005271  
82438    66.00   37.50411903197599,127.05444412993114  
82439    53.22   37.53038886115623,126.91979063028874  

[82440 rows x 9 columns]

In [52]:
df_basic_2.columns = ['item_id', 'service_type', 'residence_type', 'sales_type', 'deposit',
       'monthly_rent', 'exclusive_area', 'supply_area', 'random_location']

In [53]:
# item_id = df_basic_2['item_id'].unique()
# df_basic_2['latitude'] = 0
# df_basic_2['longitude'] = 0
# for id in item_id:
#     loc = ''.join(df_basic_2[df_basic_2['item_id']==id]['random_location']).split(',')
#     latitude, longitude = loc[0], loc[1]

#     df_basic_2.loc[df_basic_2[df_basic_2['item_id']==id].index, 'latitude'] = latitude
#     df_basic_2.loc[df_basic_2[df_basic_2['item_id']==id].index, 'longitude'] = longitude


In [54]:
# df_basic_2 = df_basic_2.drop(labels='random_location', axis=1)
df_basic_2

item_id service_type residence_type sales_type  deposit  monthly_rent  \
0      33500027         오피스텔           오피스텔         전세    13500           NaN   
1      33500123         오피스텔           오피스텔         월세     1000          65.0   
2      33500147         오피스텔           오피스텔         월세     2000         149.0   
3      33500206         오피스텔           오피스텔         전세    25000           NaN   
4      33500268         오피스텔           오피스텔         월세     2000         150.0   
...         ...          ...            ...        ...      ...           ...   
82435  34726244         오피스텔           오피스텔         월세     2000         210.0   
82436  34726267         오피스텔           오피스텔         월세     3700          76.0   
82437  34726302         오피스텔        도시형생활주택         월세     1000          65.0   
82438  34726304         오피스텔        도시형생활주택         월세      350         280.0   
82439  34726359         오피스텔           오피스텔         월세     1000          65.0   

       exclusive_area  supply_area                        random_location  
0               19.17        29.75   37.56736214187265,127.02296613394441  
1               39.67        49.59   37.49621059945249,126.87257656708832  
2               45.19       105.71    37.5662777574542,127.01694266740394  
3               28.05        37.98  37.643251120056284,127.03630435512244  
4               52.81       106.15   37.56627873736684,127.01697559113303  
...               ...          ...                                    ...  
82435           53.57        90.16   37.49389208067903,127.03072935050437  
82436           19.80        28.76  37.593366409494365,127.01872761305866  
82437           13.86        23.86   37.53088181386047,126.90751469005271  
82438           40.08        66.00   37.50411903197599,127.05444412993114  
82439           29.12        53.22   37.53038886115623,126.91979063028874  

[82440 rows x 9 columns]

### 매물 위치 정보 DataFrame

- 층 정보 어떻게 할지 고민중

In [55]:
df_loc.columns

Index(['item_id', 'local1', 'local2', 'local3', 'jibunAddress',
       'room_type_code', 'status', 'random_location', 'room_direction',
       'floor', 'floor_string', 'floor_all', 'building_id', 'Address'],
      dtype='object')

In [56]:
df_loc_2 = df_loc[['item_id', 'Address', 'local1', 'local2', 'local3', 'jibunAddress',
       'room_type_code', 'status', 'random_location', 'room_direction',
       'floor', 'floor_string', 'floor_all', 'building_id']]
df_loc_2 = df_loc_2.drop(labels='random_location', axis=1)
df_loc_2

item_id                Address local1 local2 local3     jibunAddress  \
0      33500027       서울특별시 중구 황학동 787  서울특별시     중구    황학동       중구 황학동 787   
1      33500123    서울특별시 구로구 구로동 650-4  서울특별시    구로구    구로동    구로구 구로동 650-4   
2      33500147        서울특별시 중구 흥인동 86  서울특별시     중구    흥인동        중구 흥인동 86   
3      33500206    서울특별시 도봉구 창동 630-10  서울특별시    도봉구     창동    도봉구 창동 630-10   
4      33500268        서울특별시 중구 흥인동 86  서울특별시     중구    흥인동        중구 흥인동 86   
...         ...                    ...    ...    ...    ...              ...   
82435  34726244   서울특별시 강남구 역삼동 831-45  서울특별시    강남구    역삼동   강남구 역삼동 831-45   
82436  34726267    서울특별시 성북구 동선동1가 124  서울특별시    성북구  동선동1가    성북구 동선동1가 124   
82437  34726302  서울특별시 영등포구 당산동 121-43  서울특별시   영등포구    당산동  영등포구 당산동 121-43   
82438  34726304      서울특별시 강남구 대치동 894  서울특별시    강남구    대치동      강남구 대치동 894   
82439  34726359   서울특별시 영등포구 여의도동 14-5  서울특별시   영등포구   여의도동   영등포구 여의도동 14-5   

       room_type_code  status room_direction floor floor_string floor_all  \
0                   1  cancel              W     5            5       18층   
1                   3   close             SW   NaN            고       15층   
2                   2   close              S   NaN            중       22층   
3                   4   close             SW   NaN            중       14층   
4                   2   close              S    10           10       22층   
...               ...     ...            ...   ...          ...       ...   
82435               4    open              E   NaN            저       15층   
82436               3    open              N   NaN            고        7층   
82437               2    open             NW   7.0            7       15층   
82438               4    open              S   NaN            저        7층   
82439               1    open             SW  11.0           11       13층   

       building_id  
0             2698  
1              688  
2            18926  
3            34006  
4            18926  
...            ...  
82435        18653  
82436        29919  
82437        19484  
82438        19372  
82439         3234  

[82440 rows x 13 columns]

### 옵션 정보 DataFrame

In [57]:
df_option

item_id                         options  manage_cost manage_cost_inc  \
0      33500027        01;02;03;05;08;10;11;12;          9.0             04;   
1      33500123           01;02;03;04;10;11;12;         10.0          04;05;   
2      33500147     01;02;03;05;06;07;10;11;12;         13.0          04;05;   
3      33500206           01;02;03;05;10;11;12;          7.0          04;05;   
4      33500268        01;02;03;05;06;10;11;12;         18.0          04;05;   
...         ...                             ...          ...             ...   
82435  34726244           01;02;03;05;10;11;12;         15.0             NaN   
82436  34726267        01;02;03;05;06;10;11;12;          9.0          04;05;   
82437  34726302  01;02;03;05;06;07;08;10;11;12;         13.0       03;04;05;   
82438  34726304  01;02;03;05;06;07;09;10;11;12;         18.0             NaN   
82439  34726359        01;02;03;04;09;10;11;12;         14.0          01;03;   

      manage_cost_not_inc  
0            01;02;03;05;  
1               01;02;03;  
2               01;02;03;  
3               01;02;03;  
4               01;02;03;  
...                   ...  
82435     01;02;03;04;05;  
82436           01;02;03;  
82437              01;02;  
82438     01;02;03;04;05;  
82439           02;04;05;  

[82440 rows x 5 columns]

### 매물 이미지 DataFrame

In [58]:
df_image = df[['item_id','images']].copy()
df_image

item_id                                             images
0      33500027  ['https://ic.zigbang.com/ic/items/33500027/1.j...
1      33500123  ['https://ic.zigbang.com/ic/items/33500123/1.j...
2      33500147  ['https://ic.zigbang.com/ic/items/33500147/1.j...
3      33500206  ['https://ic.zigbang.com/ic/items/33500206/1.j...
4      33500268  ['https://ic.zigbang.com/ic/items/33500268/1.j...
...         ...                                                ...
82435  34726244  ['https://ic.zigbang.com/ic/items/34726244/1.j...
82436  34726267  ['https://ic.zigbang.com/ic/items/34726267/1.j...
82437  34726302  ['https://ic.zigbang.com/ic/items/34726302/1.j...
82438  34726304  ['https://ic.zigbang.com/ic/items/34726304/1.j...
82439  34726359  ['https://ic.zigbang.com/ic/items/34726359/1.j...

[82440 rows x 2 columns]

### 부가정보 DataFrame

In [59]:
df_etc = df[['item_id', 'agent_comment', 'movein_date', 'title', 'description']].copy()
df_etc

item_id                                      agent_comment  \
0      33500027  ★ 어떤 부동산 , 어떤 중개사를 만나느냐에 따라 \r\n고객님이 편하게 안전한집을...   
1      33500123                        책임있는 중개로 늘 최선을 다하고 있습니다,,,,   
2      33500147  허위매물없이 실 매물만을 고객님께 안내해드립니다.\r\n고객이 만족하실때까지 최선을...   
3      33500206  ☆ 1억 공제증서 가입으로 안전하게 거래하실 수 있어요.\r\n☆ 가장 오래 머무르...   
4      33500268                                                NaN   
...         ...                                                ...   
82435  34726244  👍신축빌라/오피스텔/아파트 전문 서강부동산입니다. \r\n👍MBC 구해줘홈즈! 다수...   
82436  34726267  🟪 안녕하세요. 저희는 "위너부동산"입니다\r\n🟪 전원 "공인중개사자격증" 취득한...   
82437  34726302  ▫영등포 전지역 맞춤형 책임중개 약속!\r\n▫친절과 센스겸비한 중개로 시간절약 필...   
82438  34726304  ❤️ 부동산벗 공인중개사사무소 ❤️\r\n\r\n정직하고 올바른 매물 안내로 고객님...   
82439  34726359  미리 연락주셔야 약속 잡고 보실 수 있습니다.\r\n다양한 매물 여러개 있으므로  ...   

                              movein_date                            title  \
0                   2022. 11. 19 이후 입주 가능       대출 가능❤️ 전입및 확정일자가능 ❤️LHx 😍   
1            2022. 10. 14 이후 입주 가능 (조정됨.)      고대병원 후문인근\r\n\r\n SUV차 주차됨.   
2      2022. 10. 04 이후 입주 가능 (~10/10까지평일)               🧡전입가능🧡드레스룸,아일랜드식탁🧡   
3                                즉시 입주 가능       4호선 쌍문역 10분/ 2룸/ 풀옵션/ 타워주차   
4                   2022. 09. 28 이후 입주 가능  1룸 1거실,  넓은 방, 드레스룸과 거실, 전입신고가능   
...                                   ...                              ...   
82435                            즉시 입주 가능    ●내부최상급●투룸●주차편리●강남역 초역세권!●전망우수   
82436                     즉시 입주 가능 (조정필요)        🟪선착순🟪초역세권🟪신축급🟪넓은 복층형 오피스텔   
82437                            즉시 입주 가능          ❤️‍🔥보증금up,역세권,집주인짱!❤️‍🔥   
82438               2023. 01. 02 이후 입주 가능          ❤️🧡💛💚💙❤️🧡💛💚❤️🧡💛 썬앤빌.테헤란   
82439               2023. 01. 08 이후 입주 가능                 🖼상태우수 풀옵션🖼무료주차1대   

                                             description  
0      문의시에 아래 내용 남겨주세요\r\n\r\n1.입주 예정일\r\n2. 방문희망일 및...  
1      \r\n주차가 편리하고\r\n인근에  병원 시장 경찰서 구청 은행등 편의시설이\r\...  
2      💛신당역 호텔식 오피스텔\r\n💛리마크빌 A동\r\n\r\n💛신당역 12번출구 10...  
3      * 즉시입주/ 전입/ 전세 보증보험 가능\r\n\r\n* 2룸/ 화장실/ 거실 3베...  
4      신당역 바로앞 대로변 대단지 전용넓은 1룸 1거실 입니다.\r\n\r\n(일반 아파...  
...                                                  ...  
82435  ●강남역 초역세권으로, 내부 아주 깨끗하고 주차편리한 인기 투룸입니다.●\r\n\r...  
82436  \r\n💜 성신여대버스 도보 1분 거리, 성신여대입구역 도보 3분 이내 거리 위치!...  
82437  ◆◆ L O C A T I O N -  ⦋ 위 치 ⦌\r\n \r\n -당산역[2,...  
82438  ❤️ 현대썬앤빌테헤란 ❤️\r\n\r\n선릉역 역세권 도시형생활주택, 현대썬앤빌테헤...  
82439  ♣ 여의도 9호선 국회의사당역 5분 거리\r\n♣ 무료주차1대,주차하기 편한 자주식...  

[82440 rows x 5 columns]

In [23]:
# df_basic_2.to_csv(output_path+f'basic_data_{VERSION}.csv',index=False)
# df_loc_2.to_csv(output_path+f'location_data_{VERSION}.csv',index=False)
# df_option.to_csv(output_path+f'option_data_{VERSION}.csv',index=False)
# df_image.to_csv(output_path+f'image_data_{VERSION}.csv',index=False)
# df_etc.to_csv(output_path+f'etc_data_{VERSION}.csv',index=False)

In [64]:
df_loc_2['local2'].value_counts()

영등포구    11713
강서구      7765
송파구      6987
강남구      6710
동대문구     4273
구로구      4163
관악구      4162
서초구      3796
금천구      3520
마포구      3157
중구       3036
강동구      3010
서대문구     2851
종로구      2834
광진구      2536
은평구      2067
강북구      1751
중랑구      1693
성동구      1640
용산구      1619
성북구      1522
도봉구       719
동작구       415
양천구       366
노원구       135
Name: local2, dtype: int64

### VERSION 1.1

- basic_data에서 service_type 삭제

- basic_data에 동 정보 추가 : 데이터를 구 또는 동 단위로 묶어서 sequence하게 보기 위해서

        테스트를 위한 조치 : 추후 위치 기반으로 묶어서 sequence 단위로 batch를 만들 수 있음

- monthly_rent
        
        Nan값 0으로 대체

        이상치 변경


In [65]:
VERSION = '1.1'

In [66]:
df_basic_2.drop(columns='service_type', axis=1, inplace=True)

In [67]:
df_basic_3 = df_basic_2.merge(df_loc_2[['item_id','local2', 'local3']].copy(), on='item_id', how='left')
df_basic_3

item_id residence_type sales_type  deposit  monthly_rent  \
0      33500027           오피스텔         전세    13500           NaN   
1      33500123           오피스텔         월세     1000          65.0   
2      33500147           오피스텔         월세     2000         149.0   
3      33500206           오피스텔         전세    25000           NaN   
4      33500268           오피스텔         월세     2000         150.0   
...         ...            ...        ...      ...           ...   
82435  34726244           오피스텔         월세     2000         210.0   
82436  34726267           오피스텔         월세     3700          76.0   
82437  34726302        도시형생활주택         월세     1000          65.0   
82438  34726304        도시형생활주택         월세      350         280.0   
82439  34726359           오피스텔         월세     1000          65.0   

       exclusive_area  supply_area                        random_location  \
0               19.17        29.75   37.56736214187265,127.02296613394441   
1               39.67        49.59   37.49621059945249,126.87257656708832   
2               45.19       105.71    37.5662777574542,127.01694266740394   
3               28.05        37.98  37.643251120056284,127.03630435512244   
4               52.81       106.15   37.56627873736684,127.01697559113303   
...               ...          ...                                    ...   
82435           53.57        90.16   37.49389208067903,127.03072935050437   
82436           19.80        28.76  37.593366409494365,127.01872761305866   
82437           13.86        23.86   37.53088181386047,126.90751469005271   
82438           40.08        66.00   37.50411903197599,127.05444412993114   
82439           29.12        53.22   37.53038886115623,126.91979063028874   

      local2 local3  
0         중구    황학동  
1        구로구    구로동  
2         중구    흥인동  
3        도봉구     창동  
4         중구    흥인동  
...      ...    ...  
82435    강남구    역삼동  
82436    성북구  동선동1가  
82437   영등포구    당산동  
82438    강남구    대치동  
82439   영등포구   여의도동  

[82440 rows x 10 columns]

In [68]:
df_basic_3.isna().sum()

item_id                0
residence_type         0
sales_type             0
deposit                0
monthly_rent       34134
exclusive_area         0
supply_area            0
random_location        0
local2                 0
local3                 0
dtype: int64

In [69]:
df_basic_3.loc[df_basic_3[df_basic_3['monthly_rent'].isna()==True].index,'monthly_rent'] = 0

In [70]:
df_basic_3['monthly_rent'].isna().sum()

0

In [71]:
df_basic_3

item_id residence_type sales_type  deposit  monthly_rent  \
0      33500027           오피스텔         전세    13500           0.0   
1      33500123           오피스텔         월세     1000          65.0   
2      33500147           오피스텔         월세     2000         149.0   
3      33500206           오피스텔         전세    25000           0.0   
4      33500268           오피스텔         월세     2000         150.0   
...         ...            ...        ...      ...           ...   
82435  34726244           오피스텔         월세     2000         210.0   
82436  34726267           오피스텔         월세     3700          76.0   
82437  34726302        도시형생활주택         월세     1000          65.0   
82438  34726304        도시형생활주택         월세      350         280.0   
82439  34726359           오피스텔         월세     1000          65.0   

       exclusive_area  supply_area                        random_location  \
0               19.17        29.75   37.56736214187265,127.02296613394441   
1               39.67        49.59   37.49621059945249,126.87257656708832   
2               45.19       105.71    37.5662777574542,127.01694266740394   
3               28.05        37.98  37.643251120056284,127.03630435512244   
4               52.81       106.15   37.56627873736684,127.01697559113303   
...               ...          ...                                    ...   
82435           53.57        90.16   37.49389208067903,127.03072935050437   
82436           19.80        28.76  37.593366409494365,127.01872761305866   
82437           13.86        23.86   37.53088181386047,126.90751469005271   
82438           40.08        66.00   37.50411903197599,127.05444412993114   
82439           29.12        53.22   37.53038886115623,126.91979063028874   

      local2 local3  
0         중구    황학동  
1        구로구    구로동  
2         중구    흥인동  
3        도봉구     창동  
4         중구    흥인동  
...      ...    ...  
82435    강남구    역삼동  
82436    성북구  동선동1가  
82437   영등포구    당산동  
82438    강남구    대치동  
82439   영등포구   여의도동  

[82440 rows x 10 columns]

In [72]:
df_basic_3['monthly_rent'].sort_values()[-20:]

67265      1300.0
55084      1300.0
51157      1300.0
50745      1300.0
41660      1300.0
43053      1500.0
1931       2800.0
16687      5000.0
16688      5000.0
76579      6600.0
37719      7000.0
73060     28000.0
41502     31000.0
43825     45000.0
22938    195000.0
47878    581000.0
79664    710000.0
16279    750000.0
50721    915000.0
65       986000.0
Name: monthly_rent, dtype: float64

In [73]:
df_basic_3[df_basic_3['monthly_rent']>1000]

item_id residence_type sales_type  deposit  monthly_rent  \
65     33502320           오피스텔         월세    15000      986000.0   
1931   33575813           오피스텔         월세    30000        2800.0   
16279  33913106           오피스텔         월세     1000      750000.0   
16687  33919522           오피스텔         월세       50        5000.0   
16688  33919525           오피스텔         월세     5000        5000.0   
22938  33996269           오피스텔         월세    10000      195000.0   
37719  34176716           오피스텔         월세     3000        7000.0   
40256  34208050           오피스텔         월세     1300        1300.0   
41502  34223153           오피스텔         월세    13800       31000.0   
41660  34224933      아파트/업무시설          월세     1300        1300.0   
43053  34240479           오피스텔         월세     1500        1500.0   
43825  34250413           오피스텔         월세    12000       45000.0   
47878  34298514           오피스텔         월세     3000      581000.0   
50721  34332219           오피스텔         월세     1000      915000.0   
50745  34332556      아파트/업무시설          월세     1300        1300.0   
51157  34338383           오피스텔         월세     1300        1300.0   
55084  34389887      아파트/업무시설          월세     1300        1300.0   
55659  34396746           오피스텔         월세     3000        1140.0   
62874  34484471      아파트/업무시설          월세     1300        1300.0   
67265  34540493      아파트/업무시설          월세     1300        1300.0   
70819  34583750      아파트/업무시설          월세     1200        1200.0   
73060  34610180           오피스텔         월세     5000       28000.0   
74608  34629211      아파트/업무시설          월세     1200        1200.0   
76579  34653844           오피스텔         월세     2000        6600.0   
79664  34690280           오피스텔         월세     1000      710000.0   
79951  34694365      아파트/업무시설          월세     1200        1200.0   

       exclusive_area  supply_area                        random_location  \
65              29.97        46.09   37.56706467820999,127.02270178025324   
1931           192.73       490.91  37.512724332916065,127.10287257480603   
16279           18.83        34.16    37.56904678038462,127.0705046240398   
16687           23.14        26.45   37.48603361808363,126.92804594307198   
16688           23.14        26.45  37.486032301851566,126.92802016189049   
22938           13.42        22.35    37.55370932326631,126.8509735893088   
37719           18.27        27.54   37.45321073067569,126.90288394672396   
40256          130.89       281.18   37.48777169816223,127.05381504652155   
41502           26.45        33.06  37.595463532892126,127.02027612994776   
41660           99.17       168.60   37.48710829243212,127.05252809126897   
43053          121.76       274.94   37.48826274553932,127.05382122563395   
43825           28.76        33.06   37.55764654547612,127.07761179959651   
47878           16.27        29.42  37.483783891135246,126.92701192403348   
50721           24.79        36.36   37.51586227656533,127.11060584083003   
50745          130.89       193.94   37.48774715301771,127.05319670357902   
51157          130.89       281.18  37.488243910682286,127.05384774997538   
55084          130.89       193.94   37.48831949595818,127.05381512986044   
55659           29.75        69.03  37.515752182844615,127.11263621337494   
62874          130.89       193.94  37.488321422587525,127.05317839721178   
67265          130.89       193.94    37.48783815180496,127.0531566811437   
70819          130.89       193.94   37.48783116642514,127.05315106670467   
73060           24.17        52.35  37.633697697029156,126.91834100201119   
74608          130.89       193.94   37.48830632247415,127.05315985042212   
76579           21.49        36.36   37.48005437548006,126.95386492138195   
79664           16.53        26.45   37.48436748859769,126.95528103264151   
79951          130.89       193.94  37.487848855998074,127.05314930070185   

      local2 local3  
65        중구    황학동  
1931     송파구    신천동  
16279   동대문구    장안동  
166

In [74]:
# 이상치 변경
df_basic_3.loc[65,'monthly_rent']*=1e-4
df_basic_3.loc[1931,'monthly_rent']*=1e-1
df_basic_3.loc[16279,'monthly_rent']*=1e-4

df_basic_3.loc[22938,'monthly_rent']*=1e-4
df_basic_3.loc[37719,'monthly_rent']*=1e-2
df_basic_3.loc[41502,'monthly_rent']*=1e-4
df_basic_3.loc[43825,'monthly_rent']*=1e-3

df_basic_3.loc[47878,'monthly_rent']*=1e-4
df_basic_3.loc[50721,'monthly_rent']*=1e-4
df_basic_3.loc[55659,'monthly_rent']*=1e-1
df_basic_3.loc[73060,'monthly_rent']*=1e-3
df_basic_3.loc[76579,'monthly_rent']*=1e-2

df_basic_3.loc[79664,'monthly_rent']*=1e-4

df_basic_3.loc[16688,'monthly_rent']*=1e-2
df_basic_3.loc[50722,'monthly_rent']*=1e-1

In [75]:
a = df_basic_3.loc[16687,'deposit']
b = df_basic_3.loc[16687,'monthly_rent']
df_basic_3.loc[16687,'deposit'] = b
df_basic_3.loc[16687,'monthly_rent'] = a

a = df_basic_3.loc[32858,'deposit']
b = df_basic_3.loc[32858,'monthly_rent']
df_basic_3.loc[32858,'deposit'] = b
df_basic_3.loc[32858,'monthly_rent'] = a

In [76]:
df_basic_3.drop_duplicates(subset='random_location', inplace=True)

In [77]:
df_basic_3

item_id residence_type sales_type  deposit  monthly_rent  \
0      33500027           오피스텔         전세    13500           0.0   
1      33500123           오피스텔         월세     1000          65.0   
2      33500147           오피스텔         월세     2000         149.0   
3      33500206           오피스텔         전세    25000           0.0   
4      33500268           오피스텔         월세     2000         150.0   
...         ...            ...        ...      ...           ...   
82435  34726244           오피스텔         월세     2000         210.0   
82436  34726267           오피스텔         월세     3700          76.0   
82437  34726302        도시형생활주택         월세     1000          65.0   
82438  34726304        도시형생활주택         월세      350         280.0   
82439  34726359           오피스텔         월세     1000          65.0   

       exclusive_area  supply_area                        random_location  \
0               19.17        29.75   37.56736214187265,127.02296613394441   
1               39.67        49.59   37.49621059945249,126.87257656708832   
2               45.19       105.71    37.5662777574542,127.01694266740394   
3               28.05        37.98  37.643251120056284,127.03630435512244   
4               52.81       106.15   37.56627873736684,127.01697559113303   
...               ...          ...                                    ...   
82435           53.57        90.16   37.49389208067903,127.03072935050437   
82436           19.80        28.76  37.593366409494365,127.01872761305866   
82437           13.86        23.86   37.53088181386047,126.90751469005271   
82438           40.08        66.00   37.50411903197599,127.05444412993114   
82439           29.12        53.22   37.53038886115623,126.91979063028874   

      local2 local3  
0         중구    황학동  
1        구로구    구로동  
2         중구    흥인동  
3        도봉구     창동  
4         중구    흥인동  
...      ...    ...  
82435    강남구    역삼동  
82436    성북구  동선동1가  
82437   영등포구    당산동  
82438    강남구    대치동  
82439   영등포구   여의도동  

[82440 rows x 10 columns]

### Save

In [78]:
df_basic_3.to_csv(output_path+f'basic_data_{VERSION}.csv',index=False)